src/pipes.cpp

```diff
@@ -0,0 +1,66 @@
+#include "pipes.h"
+
+const int Pipes::NUM_PIPES = 4;
+const float Pipes::PIPE_SCALE = 0.4;
+const int Pipes::PIPE_OFFSET_RANGE = 300;
+const float Pipes::SPEED_PER_SECOND = 500;
+
+Pipes::Pipes(int screenWidth, int screenHeight, int frameRate)
+{
+    bool res = pipeTexture.loadFromFile("./lib/pipe.png");
+    if(!res) throw("Cannot load pipe\n");
+
+    pipeSprites = new sf::Sprite[NUM_PIPES];
```

Comment:
Ew wtf why are you using `new []`... just make this a `std::vector<sf::Sprite>`. `std::vector` is the same as a java `ArrayList`...

I would honestly reject your PR from this single line alone.